lib/android/app/src/main/java/com/reactnativenavigation/anim/NavigationAnimator.java

```diff
@@ -0,0 +1,192 @@
+        view.setVisibility(View.INVISIBLE);
+        ObjectAnimator alpha = ObjectAnimator.ofFloat(view, View.ALPHA, 0, 1);
+        alpha.setInterpolator(DECELERATE_INTERPOLATOR);
+
+        AnimatorSet set = new AnimatorSet();
+        set.addListener(new Animator.AnimatorListener() {
+            @Override
+            public void onAnimationStart(Animator animation) {
+                view.setVisibility(View.VISIBLE);
+            }
+
+            @Override
+            public void onAnimationEnd(Animator animation) {
+                if (animationListener != null) {
+                    animationListener.onAnimationEnd();
+                }
+            }
+
+            @Override
+            public void onAnimationCancel(Animator animation) {
+
+            }
+
+            @Override
+            public void onAnimationRepeat(Animator animation) {
+
+            }
+        });
+        ObjectAnimator translationY = ObjectAnimator.ofFloat(view, View.TRANSLATION_Y, this.translationY, 0);
+        translationY.setInterpolator(DECELERATE_INTERPOLATOR);
+        translationY.setDuration(DURATION);
+        alpha.setDuration(DURATION);
+        set.playTogether(translationY, alpha);
+        set.start();
+    }
+
+    public void animatePop(View view, @Nullable final StackAnimationListener animationListener) {
+        ObjectAnimator alpha = ObjectAnimator.ofFloat(view, View.ALPHA, 1, 0);
+        alpha.setInterpolator(ACCELERATE_INTERPOLATOR);
+
+        AnimatorSet set = new AnimatorSet();
+        set.addListener(new Animator.AnimatorListener() {
+            @Override
+            public void onAnimationStart(Animator animation) {
+
+            }
+
+            @Override
+            public void onAnimationEnd(Animator animation) {
+                if (animationListener != null) {
+                    animationListener.onAnimationEnd();
+                }
+            }
+
+            @Override
+            public void onAnimationCancel(Animator animation) {
+
+            }
+
+            @Override
+            public void onAnimationRepeat(Animator animation) {
+
+            }
+        });
+        ObjectAnimator translationY = ObjectAnimator.ofFloat(view, View.TRANSLATION_Y, 0, this.translationY);
+        translationY.setInterpolator(ACCELERATE_INTERPOLATOR);
+        translationY.setDuration(DURATION);
+        alpha.setDuration(DURATION);
+        set.playTogether(translationY, alpha);
+        set.start();
+    }
+
+    private float getWindowHeight(Context context) {
+        DisplayMetrics metrics = new DisplayMetrics();
+        WindowManager windowManager = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
+        if (windowManager != null) {
+            windowManager.getDefaultDisplay().getMetrics(metrics);
+        }
+        return metrics.heightPixels;
+    }
+
+    public void animateShowTopBar(final TopBar topBar, final View container) {
```

Comment:
Seems like NavigationAnimator has two concerns and should be split into two classes.
1. Animating screens
2. Animating TopBar

Reply:
I really dislike the word `container`... we abused it a lot and would love phasing it out.
Please rename to `contentView` which is also the naming used in `NavigationOptions`